### PR TITLE
DAOS-7821 build: Overhaul the github actions integration.

### PR DESCRIPTION
**Pull request by ashleypittman**

### CODE REVIEW

File: .github/workflows/ci2.yml [hunk 1]
```diff
@@ -90,48 +58,83 @@ jobs:
...
-        # Run this on distros where we don't run the ci1 Workflow.
-        distro: [ubuntu.20.04, centos.8, leap.15]
+        distro: [ubuntu.20.04, centos.8, leap.15, fedora, rocky, alma]
         compiler: [clang, gcc]
+        include:
+          - distro: fedora
+            base: centos.8
+            with: fedora:34
+            java: NO
+          - distro: rocky
```
Comment (Michael-Hennecke) [inline .github/workflows/ci2.yml:72]: given that RedHat has just released 9.0-beta, should we be prepared for that and rename rocky to rocky.8 and alma to alma.8 ?

File: utils/docker/Dockerfile.centos.8 [hunk 2]
```diff
@@ -187,17 +189,17 @@ RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
...
 
-# force an upgrade to get any newly built RPMs
+# force an upgrade to get any newly built RPMs, but only if CACHEBUST is set.
 USER root:root
 ARG CACHEBUST
-RUN dnf -y upgrade \
-        --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
-    dnf clean all
+RUN [ -z "$CACHEBUST" ] || \
+    { dnf upgrade --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
```
Comment (Michael-Hennecke) [inline utils/docker/Dockerfile.centos.8:196]: not dnf -y ?

File: utils/scripts/install-centos8.sh [hunk 3]
```diff
@@ -10,10 +10,11 @@
 
 set -e
 
-dnf -y upgrade
-dnf -y install \
+dnf upgrade
```
Comment (Michael-Hennecke) [inline utils/scripts/install-centos8.sh:13]: not dnf -y (2 times)?

File: .github/workflows/ci2.yml (hunk 1 above)
Comment (ashleypittman) [reply .github/workflows/ci2.yml:72]: We should be prepared to update all the upstream versions we're tracking, ideally that should be in a self-contained PR however, and having this code tidy and consistent would help with that.

File: utils/docker/Dockerfile.centos.8 (hunk 2 above)
Comment (ashleypittman) [reply utils/docker/Dockerfile.centos.8:196]: The "dnf config-manager --save --setopt=assumeyes=True" above applies this in the config file so we can remove all the -y and --assumeyes options from there forward.

**Review by Michael-Hennecke — APPROVED** [review]

**Pull request merged**